__tests__/app.test.js

```diff
@@ -7,6 +7,36 @@ const data = require('../db/data/test-data');
 beforeEach(() => seed(data));
 afterAll(() => db.end());
 
+describe('/api', () =>
+{
+    describe('GET', () =>
+    {
+        test("STATUS 200 - Responds with a JSON object describing all the available endpoints.", () =>
+        {
+            return request(app)
+                .get('/api')
+                .expect(200)
+                .then(({ body: { endpoints } }) =>
+                {
+                    parsedEndpoints = JSON.parse(endpoints);
+
+                    // 'GET /api' only has the 'description' property.
+                    expect(parsedEndpoints['GET /api'].description).toBe('Serves up a JSON representation of all the available endpoints of the API.');
+
+                    // Otherwise, rest should have all 4 properties: 'description', 'queries', 'requestFormat', and 'exampleResponse'.
+                    const remainingEndpoints = Object.values(parsedEndpoints).slice(1, Object.values(parsedEndpoints).length);
+                    remainingEndpoints.forEach(({ description, queries, requestFormat, exampleResponse }) =>
+                    {
+                        expect(typeof description).toBe('string');
+                        expect(Array.isArray(queries)).toBe(true);
+                        expect(typeof requestFormat).toBe('object');
+                        expect(typeof exampleResponse).toBe('object');
+                    });
+                });
+        });
```

Comment:
This is an interesting way to make this test dynamic, but there is a nicer way to do this. For this endpoint we simply want to assert that what we get back is the endpoints.json file contents, we dont really care on the test side what this contains, so long as it matches the endpoints.json file. Can you think of a way to do this, so that the response contains whatever the contents of the endpoints.json file has?